Add a couple of enhancements to the local testing guide. (#263)
python-discord/infra: ansible/local_testing/README.md

# Testing Locally

This section contains the necessary steps in order to be able to setup virtual machines that mimic our netcup servers and be able to run the ansible roles/playbooks we have on them.

### Requirements

- [Vagrant](https://developer.hashicorp.com/vagrant/docs/installation)
- [VirtualBox](https://www.virtualbox.org/wiki/Downloads)

### Getting started

1. Creating the virtual boxes

On your host machine, run the following:

```bash
cd ansible/local_testing
vagrant up
```

This will create all the virtual boxes on which we will run our ansible playbooks.

2. Push the control VM's ssh key to all nodes

```bash
vagrant ssh  # This will give you terminal access to the `control` VM, which we will use as the ansible control node.

cd /home/vagrant/infra
bash ./local_testing/scripts/push-keys
```

> If you're on windows, you might run into line ending issues when running the `push-keys` script.

> > To fix it, run this **while still inside the ssh session*: `sed -i 's/\r$//' /home/vagrant/infra/local_testing/scripts/push-keys`

Reviewer: I'm not on Windows, but shouldn't the file be checked in without `\r` line endings and thus not cause any issues?
Although it might still, because it's Windows. I'm not going to act like it makes sane decisions

Author: That's the theory indeed, but when I tried  to push the SSH keys it started complaining :/

Should we maybe add it to .gitannotations ?

Reviewer: Hmm, I think that's worth a shot! If we can make the local setup easier
in any way we should.

# Fixes/Notes

### Ansible cannot decrypt the files encrypted with ansible vault.

> * The `ansible\roles\certbot\vars\main.yaml` and `ansible\roles\pydis-users\vars\main.yaml` files have been encrypted with ansible vault due to their sensitive content.

Reviewer: Should we use forward slashes here? On both GitHub and any non-Windows system those are used, so you could also e.g. link to them by using forward slashes. I would find it better for consistency

Author: Sounds good to me, I'll make the change

> * If you're not part of the people with access to the vault, you're going to have to either define your own variables or just exclude the appropriate roles.

Reviewer: Maybe we can make this a bit more concise:
```suggestion
* If you lack access to the vault, you're going to have to either define your own variables or exclude the appropriate roles.
```

Author: Simple & elegant, granted
